nomad/grapl-core.nomad

```diff
@@ -724,14 +746,16 @@ job "grapl-core" {
       driver = "docker"
 
       config {
-        image        = "${var.container_registry}grapl/node-identifier:${var.node_identifier_tag}"
-        network_mode = "grapl-network"
+        image = "${var.container_registry}grapl/${var.container_repo}node-identifier:${var.node_identifier_tag}"
```

Comment:
This funky interpolation is done so we can support both the local grapl format (grapl/container-name:tag) and the cloudsmith format (registry/org/repo/container-name:tag)

Reply:
While not necessarily a change for now, I wonder if it might be better in the long term to just have this refer to some `${var.node_identifier_image}` (and similar for every other service), and pull the composition of the name up to a higher level. This is what happened when Pulumi set up our Fargate jobs before Nomad, for instance.

If things are local, we're generally just looking for `grapl/node-identifier:latest` (or whatever).

In real deployments, however, we'll have a Cloudsmith URL, which (as you point out) is a bit different, like `docker.cloudsmith.io/grapl/raw/node-identifier:20210806001956-3bd9525c`. Currently the `raw` bit is just a catch-all repository that all the containers go into, but longer term, we'll have additional repositories that we'll promote containers through as they go through the pipeline, such that production releases will be using something like `docker.cloudsmith.io/grapl/release/node-identifier:20210806001956-3bd9525c`.

Ultimately, we shouldn't be embedding that logic deep down in each job.

Reply:
If you're ok with it, I'd like to tackle this in a later PR. I do agree that I'd prefer to pass in a full image from Pulumi.

Reply:
Another PR is great; would you mind filing an issue for it so we don't lose track of it?

Reply:
https://app.zenhub.com/workspaces/grapl-6036cbd36bacff000ef314f2/issues/grapl-security/issue-tracker/697